app/validators/uuid_validator.rb

```diff
@@ -0,0 +1,13 @@
+class UuidValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless valid_uuid?(value)
+      record.errors.add(attribute, "is not a valid UUID")
+    end
+  end
+
+private
+
+  def valid_uuid?(uuid)
+    uuid =~ /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/
```

Comment:
I'd suggest setting this regex as a constant and then you can avoid the need for this particular method and do the check in the `validate_each` method. I'd also suggest using the regex from https://github.com/alphagov/email-alert-api/blob/b6428880aa730e316803d7129db3ec47304e933b/app/models/message.rb#L11 as it uses `\A` and `\Z` start and end of string (see https://stackoverflow.com/questions/577653/difference-between-a-z-and-in-ruby-regular-expressions for more).

Reply:
Updated as requested.